python_modules/dagster/dagster/_core/definitions/sensor_definition.py

```diff
@@ -306,7 +542,9 @@ def advance_cursor(
         """
         cursor_dict = json.loads(self.cursor) if self.cursor else {}
         update_dict = {
-            str(k): v.storage_id if v else cursor_dict.get(str(k))
+            str(k): (self.get_partition_from_event_log_record(v), v.storage_id)
```

Comment:
one of the things i struggled with the most when i was writing this Context type was keeping track of what info the cursor contained and what the types of the keys/values were. now that the cursor is more complicated, i think it would be really helpful  for whoever works on this next to have a comment somewhere in the class that describes the format of the cursor and what info it contains. something like

```
The cursor maps stringified AssetKeys (str(asset_key)) to Tuples. The first value in the tuple is the partition of type X which means Y. The second value in the tuple is the storage_id (int) of the materialization event for the asset that has been most recently fetched from the DB
```

Reply:
Why does the cursor need to include the partition key? In theory we could get it by pulling the event using the storage ID? Or is it an optimization?

Reply:
It's an optimization--if a user wants to see what partition key the cursor is on, storing the partition key of the cursor allows us to avoid refetching the event log record from storage.